### PR TITLE
Fix docc archive bugs

### PR DESCRIPTION
**Pull request by AttilaTheFun**
This PR builds upon and fixes @luispadron 's draft PR:
https://github.com/bazelbuild/rules_apple/pull/2583

Which fixes https://github.com/bazelbuild/rules_apple/issues/2582

docc preview supports previewing with or without a .docc bundle. There were some issues in the argument handling of the template script that stopped this from working when no bundle was provided. This fixes it and additionally fixes the following warning:

warning: '--index' is deprecated. Use '--emit-lmdb-index' indead.
 docc preview will emit a json representation by default and we do not need the lmdb index afaict.

Additionally, previously the files within a .docc directory were not included as inputs to the action, so it would not be properly invalidated if you changed the directory contents. 

This PR includes all of the files within the data attribute under the .docc directory.

### CODE REVIEW

File: test/docc_archive_test.sh
Comment (luispadron) [inline test/docc_archive_test.sh:1]: If you remove this test CI should pass. I can follow up on seeing what it takes to fix, some issue with this test runner change the run path.

**Review by luispadron — APPROVED** [review]
Thank you!

File: apple/internal/docc.bzl [hunk 1]
```diff
@@ -258,6 +262,10 @@ This value must be either `fileprivate`, `internal`, `private`, or `public`. The
                 allow_single_file = True,
                 default = "//apple/internal/templates:docc_preview_template",
             ),
+            "_bash_runfiles": attr.label(
+                allow_files = True,
+                default = "@bazel_tools//tools/bash/runfiles",
+            ),
```
Comment (luispadron) [inline apple/internal/docc.bzl:268]:
```suggestion
```

Leftover from my PR when testing but I don't think it's needed

File: test/docc_archive_test.sh
Comment (AttilaTheFun) [reply test/docc_archive_test.sh:1]: Ok! Yeah it was timing out after five minutes for me.

**Pull request merged**